client/components/tinymce/plugins/wpcom-track-paste/plugin.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import tinymce from 'tinymce/tinymce';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+const debug = debugFactory( 'calypso:tinymce-plugins:wpcom-paste-track' );
+
+const SOURCE_GOOGLE_DOCS = 'google_docs';
+const SOURCE_UNKNOWN = 'unknown';
+
+function trackPaste( editor ) {
+	debug( 'init' );
+
+	const store = editor.getParam( 'redux_store' );
+	if ( ! store ) {
+		debug( 'store not found, do not bind events' );
+		return;
+	}
+
+	/**
+	 * Records the event.
+	 *
+	 * @param {String} mode - 'html-editor' or 'visual-editor', indicates which editor was in use on paste.
+	 * @param {(Array|DOMStringList)} types - The types the content is available to paste.
+	 */
+	function recordPasteEvent( mode, types ) {
+		debug( 'track paste event ' );
+		const typesStr = toString( types );
+		store.dispatch( recordTracksEvent( 'calypso_editor_content_paste', {
+			mode,
+			types: typesStr,
+			source: getSource( typesStr )
+		} ) );
+	}
+
+	function isGoogleDocs( types ) {
+		return ( types.indexOf( 'application/x-vnd.google-docs-image-clip+wrapped' ) > -1 ) ||
+		( types.indexOf( 'application/x-vnd.google-docs-document-slice-clip+wrapped' ) > -1 );
```

Comment:
Pulling all these libraries for a simple operation like this seemed to me quite excessive. But I agree the code could be more readable, so I ended up refactoring it and converted the `isGoogleDocs( types ) ` param to an array (it was an string before), so we have a much more semantic code.